src/jsp/jsp_sr.c

```diff
@@ -356,12 +356,13 @@ jsp_start_server (const char *db_name, const char *path)
   jint res;
   jclass cls, string_cls;
   jmethodID mid;
-  jstring jstr_dbname, jstr_path, jstr_version, jstr_envroot;
+  jstring jstr_dbname, jstr_path, jstr_version, jstr_envroot, jstr_port;
   jobjectArray args;
   JavaVMInitArgs vm_arguments;
   const int vm_n_options = 3;
   JavaVMOption options[vm_n_options];
   char classpath[PATH_MAX + 32], logging_prop[PATH_MAX + 32];
+  char port[5];
```

Comment:
port[6] seems to correct because it needs to include null character. The max port number is 65535.